high_card.rb

```diff
@@ -1,5 +1,117 @@
 # Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
-players = []
+$winning_card = nil
+$second_winning_card = nil
+$players = []
+
+$r_ordered = [2,3,4,5,6,7,8,9,10,"J","Q","K","A"]
+# 0 to 12
+$s_ordered = ["clubs", "diamonds", "hearts", "spades"]
+# 0 to 3
+
+def shuffled_deck
+
+  new_deck = []
+  $r_ordered.each_with_index do |rank, index|
+    $s_ordered.each_with_index do |suit, index_two|
+      new_deck.push({
+        rank_n: index,
+        rank: rank,
+        suit_n: index_two,
+        suit: suit
+        })
+      end
+    end
+
+    shuffled = new_deck.shuffle
+    return shuffled
+  end
+
```

Comment:
great constructor method, I like your use of double `each_with_index` to save both the rank's and suit's 'value'!